### PR TITLE
Append strapi.dir to plugin path if relative

### PR DESCRIPTION
**Pull request by petersg83**

### CODE REVIEW

File: packages/core/strapi/lib/core/loaders/plugins/get-enabled-plugins.js [hunk 1]
```diff
@@ -9,6 +9,8 @@ const loadConfigFile = require('../../app-configuration/load-config-file');
 
 const isStrapiPlugin = info => get('strapi.kind', info) === 'plugin';
 
+const isPath = path => /^(\.\/|\.\.\/|\/|\.\\|\.\.\\|\\)/.test(path);
```
Comment (Elhebert) [inline packages/core/strapi/lib/core/loaders/plugins/get-enabled-plugins.js:12]: You could simplify this by doing: 
```js
const isPath = path => /^(\.{0,2}(\/|\\))/.test(path);
```

You can also use `path.sep` to get the separator used by the server: 
```js
const isPath = path => new RegExp('^(.{0,2}\\' + path.sep + ')').test(path);
```

File: packages/core/strapi/lib/core/loaders/plugins/get-enabled-plugins.js [hunk 2]
```diff
@@ -26,12 +28,19 @@ const toDetailedDeclaration = declaration => {
...
-      pathToPlugin = dirname(require.resolve(declaration.resolve));
-    } catch (e) {
-      if (existsSync(declaration.resolve) && statSync(declaration.resolve).isDirectory()) {
-        pathToPlugin = declaration.resolve;
-      } else {
+
+    if (isPath(declaration.resolve)) {
+      pathToPlugin = isAbsolute(declaration.resolve)
+        ? declaration.resolve
+        : join(strapi.dir, declaration.resolve);
```
Comment (Elhebert) [inline packages/core/strapi/lib/core/loaders/plugins/get-enabled-plugins.js:35]: You could use [`path.resolve`](https://nodejs.org/api/path.html#path_path_resolve_paths) to get rid of the ternary conditional:

```js
pathToPlugin = path.resolve(strapi.dir, declaration.resolve);
```

If `declaration.resolve` is an absolute path, it'll be used as is. If it's a relative one, it'll be relative to `strapi.dir`.

Comment (petersg83) [reply packages/core/strapi/lib/core/loaders/plugins/get-enabled-plugins.js:35]: Good idea

File: packages/core/strapi/lib/core/loaders/plugins/get-enabled-plugins.js [hunk 3]
```diff
@@ -26,12 +26,17 @@ const toDetailedDeclaration = declaration => {
...
   if (has('resolve', declaration)) {
     let pathToPlugin = '';
-    try {
-      pathToPlugin = dirname(require.resolve(declaration.resolve));
-    } catch (e) {
-      if (existsSync(declaration.resolve) && statSync(declaration.resolve).isDirectory()) {
-        pathToPlugin = declaration.resolve;
-      } else {
+
+    if (isPath(declaration.resolve)) {
```
Comment (alexandrebodin) [inline packages/core/strapi/lib/core/loaders/plugins/get-enabled-plugins.js:30]: How do we handle path like `C:\some-dir\file` ? 

Would it work if we did: try a simple require.resolve and then a require.resolve with the path.resolve(strapi.dir, value) ?

Comment (petersg83) [reply packages/core/strapi/lib/core/loaders/plugins/get-enabled-plugins.js:30]: Good point, I'll think of that

Comment (petersg83) [reply packages/core/strapi/lib/core/loaders/plugins/get-enabled-plugins.js]: Figuring out if the string is more likely a path or a npm module seems too tricky.
For the moment I went back to the system of before : trying to resolve as module first and, if it doesn't find it, try to resolve as a path (append with strapi.dir if necessary).
Conflict should be rare.
Do you have a better idea?

**Review by alexandrebodin — APPROVED** [review]

**Pull request merged**